docs/en/advanced_guides/datasets.md

```diff
@@ -1 +1,247 @@
-# Datasets
+# Dataset
+
+Due to the Dataset class in each OpenMMLab codebases have some common characteristics and requirements, such as efficient internal data storage format, support for the concatenation of different datasets, dataset repeated sampling,
+MMEngine implements `BaseDataset` which provides some basic interfaces and implements some DatasetWrappers with the same interfaces. The Dataset class of each OpenMMLab downstream codebase would inherit this base dataset class and add customization methods.
+
+The basic function of the MMEngine `BaseDataset` is loading the dataset information which has two categories.
+One is meta information, which represents the information related to the dataset itself and sometimes needs to be obtained by the model or other external components.
+The other is data information, which defines the file path and corresponding label information of specific data info.
+
+More details of `BaseDataset` could be found in [MMEngine BaseDataset Documentation](https://github.com/open-mmlab/mmengine/blob/main/docs/en/advanced_tutorials/basedataset.md).
+
+## BaseSegDataset
+
+`BaseSegDataset` is a base dataset class for semantic segmentation task in MMSegmentation v1.x, which is inherited from `BaseDataset` in MMEngine.
+
+In this document, we will introduce `BaseSegDataset` class in MMSegmentation.
+First, we would introduce standard dataset format in MMSegmentation, while creating a corresponding dataset class is also necessary.
+Then, we would introduce some important methods of `BaseSegDataset`: (1) initialization of `BaseSegDataset`, including detailed methods which would be called during this process.
+(2) loading meta information and (3) loading raw data information, the loading function of these two data information categories are basic function of `BaseSegDataset` for training and test process.
+Moreover, (4) `__getitem__` method used to give index of data information in `BaseSegDataset` initialization, which is similar to `torch.utils.data.Dataset`.
+
+More information about `BaseSegDataset` could be found in [basesegdataset.py](https://github.com/open-mmlab/mmsegmentation/blob/1.x/mmseg/datasets/basesegdataset.py).
+
+### Standard Dataset format
+
+All datasets such as `CityscapesDataset` and `ADE20KDataset` are inherited from `BaseSegDataset` in the 1.x version of MMSegmentation.
+Some common methods are defined in `BaseSegDataset`,
+such as loading data information including meta information like categories and palette of dataset,
+and data information like path of image and ground truth, and modifying labels of dataset.
+
+`BaseSegDataset` is designed for supervised semantic segmentation task thus both images and annotations are necessary when model training.
+The directory structure of corresponding dataset is as below.
```

Comment:
I think this section is about dataset format, so please move the 'Some common methods ....' to proper position